python/graphframes/graphframe.py

```diff
@@ -320,12 +322,18 @@ def parallelPersonalizedPageRank(self, resetProbability = 0.15, sourceIds = None
         :param maxIter: the fixed number of iterations this algorithm runs
         :return:  GraphFrame with new vertices column "pageranks" and new edges column "weight"
         """
-        assert sourceIds is not None, "Source vertices Ids sourceIds must be provided"
+        assert sourceIds is not None and len(sourceIds) > 0, "Source vertices Ids sourceIds must be provided"
         assert maxIter is not None, "Max number of iterations maxIter must be provided"
+        if isinstance(sourceIds, list):
+            _gateway = self._sc._gateway
+            _srcIds = _gateway.new_array(_gateway.jvm.Object, len(sourceIds))
```

Comment:
The `Scala API` expects `sourceIds` to be of type `Array[Any]`. 
Directly passing a `Python list` triggers an error. 
We explicitly create a `Java Array` and copy the elements to it.